src/Eff/Unit.cs

```diff
@@ -9,21 +9,15 @@ namespace Nessos.Effects
     ///   While similar to <see cref="Void"/>, the unit type is inhabited 
     ///   by a single materialized value and can be used as a generic parameter.
     /// </remarks>
-    public readonly struct Unit : IEquatable<Unit>
+    public readonly record struct Unit
     {
         /// <summary>
         ///   Gets the Unit instance.
         /// </summary>
         public static Unit Value => new();
         /// Implements unit hashcode
         public override int GetHashCode() => 1;
```

Comment:
Isn't the GetHashCode() override redundant as well now?

Reply:
Yes it is, but I kept it for compatibility. The compiler-generated version returns 0. See [my opening comment](https://github.com/nessos/Eff/pull/34#issue-2430448014):

> I've kept `GetHashCode` so that it continues to return 1. I doubted anyone is relying on it, but it can help avoid conflicts (in hashing buckets) if someone is using `Unit?` as keys, where hash of a `default(Unit?)` will return 0 and 1 if a `Unit?` is some unit. If this is not needed, and once again I doubted anyone is counting on that, I can also remove it.

Should I remove it?

Reply:
Done with b36ab83d00d9dcc228449b42bffbb203d55dbe73.